synapse/handlers/presence.py

```diff
@@ -755,6 +772,11 @@ def run_persister() -> Awaitable[None]:
         self._event_pos = self.store.get_room_max_stream_ordering()
         self._event_processing = False
 
+        # The per-device presence state, maps user to devices to per-device presence state.
+        self.user_to_device_to_current_state: Dict[
+            str, Dict[Optional[str], UserDevicePresenceState]
+        ] = {}
```

Comment:
We don't currently remove things from this dictionary which means that once a user goes "online" it probably can't idle back to unavailable right now. (I need to add a test for this.) This is because we would consider the last activity from e.g. an unavailable sync loop, but the online state of a no-longer existing sync loop.

Reply:
I fixed this so that things are now timed out.